src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -622,273 +422,208 @@ try
+    //                         | topSideBearing       |  1/2 lineGap    |
+    // aaaaaa   ggggggg     <-------------------------|-------------|   |
+    //      a   g    g                                |             |   |
+    //  aaaaa   ggggg                                 |<-ascent     |   |
+    // a    a   g                                     |             |   |---- lineHeight
+    // aaaaa a  gggggg      <----baseline, verticalOriginY----------|---|
+    //          g     g                               |<-descent    |   |
+    //          gggggg      <-------------------------|-------------|   |
+    //                         | bottomSideBearing    | 1/2 lineGap     |
+    // -----------------    <===========================================|
+    //
+    // aaaaaa   ggggggg     top of next line
+    // ...
+    //
+    // Also note...
+    // We're going to add half the line gap to the ascent and half the line gap to the descent
+    // to ensure that the spacing is balanced vertically.
+    // Generally speaking, the line gap is added to the ascent by DirectWrite itself for
+    // horizontally drawn text which can place the baseline and glyphs "lower" in the drawing
+    // box than would be desired for proper alignment of things like line and box characters
+    // which will try to sit centered in the area and touch perfectly with their neighbors.
+
+    const auto fullPixelAscent = ceil(ascent + halfGap);
+    const auto fullPixelDescent = ceil(descent + halfGap);
+    lineSpacing.height = fullPixelAscent + fullPixelDescent;
+    lineSpacing.baseline = fullPixelAscent;
+
+    // According to MSDN (https://docs.microsoft.com/en-us/windows/win32/api/dwrite_3/ne-dwrite_3-dwrite_font_line_gap_usage)
+    // Setting "ENABLED" means we've included the line gapping in the spacing numbers given.
+    lineSpacing.fontLineGapUsage = DWRITE_FONT_LINE_GAP_USAGE_ENABLED;
+
+    _lineSpacing = lineSpacing;
+
+    // The scaled size needs to represent the pixel box that each character will fit within for the purposes
+    // of hit testing math and other such multiplication/division.
+    COORD coordSize = { 0 };
+    coordSize.X = gsl::narrow<SHORT>(widthExact);
+    coordSize.Y = gsl::narrow_cast<SHORT>(lineSpacing.height);
+
+    // Unscaled is for the purposes of re-communicating this font back to the renderer again later.
+    // As such, we need to give the same original size parameter back here without padding
+    // or rounding or scaling manipulation.
+    const COORD unscaled = desired.GetEngineSize();
+
+    const COORD scaled = coordSize;
+
+    actual.SetFromEngine(_defaultFontInfo.GetFamilyName(),
+                         desired.GetFamily(),
+                         DefaultTextFormat()->GetFontWeight(),
+                         false,
+                         scaled,
+                         unscaled);
+
+    actual.SetFallback(_defaultFontInfo.GetFallback());

```

Comment:
Seeing both `SetFromEngine` and `SetFallback` is a bit weird but for now I can live with it. It matches the old code. 

We can probably leave it to some kids in the future.

Reply:
Sure